DataMigration/DataLayer/PostgresConnect.cs

```diff
@@ -0,0 +1,202 @@
+                using (var command = GetCommand(conn, "select count(*) from historical_ocr_data"))
+                {
+                    using (var reader = command.ExecuteReader())
+                    {
+                        while (reader.Read())
+                        {
+                            res = reader.GetInt32(0);
+                        }
+                    }
+                }
+            }
+            return res;
+        }
+        public List<string> GetAllHistoricPaths(string connectString,int limit)
+        {
+            var docs=new List<string>();
+            try
+            {
+                _log.WriteLog(LogLevel.Info, $"Get {limit} historic paths (Postgres)\n");
+                using (var conn = new NpgsqlConnection(connectString))
+                {
+                    conn.Open();
+                    using (var command = GetCommand(conn, $"SELECT fullfilepath FROM historical_ocr_data LIMIT {limit}"))
+                    {
+                        using (var reader = command.ExecuteReader())
+                        {
+                            while (reader.Read())
+                            {
+                                docs.Add(ReplaceUnsupportedCharacters(reader.GetString(0), "/", "\\"));
+                            }
+                        }
+                    }
+                }               
+            }
+            catch (Exception ex)
+            {
+                _log.WriteLog(LogLevel.Info, "Something wrong with executing command or with data in table \n");
+                _log.WriteLog(LogLevel.Error, ex.Message + "\n" + ex.StackTrace);
+            }
+            return docs;
+        }
+        public NpgsqlCommand GetCommand(NpgsqlConnection connection, string commandText)
+        {
+            return new NpgsqlCommand
+            {
+                Connection = connection,
+                CommandTimeout = 0,
+                CommandType = CommandType.Text,
+                CommandText = commandText
+            };
+        }
+        public void InsertOcr(string connectString,List<HistoricalOcrData> historicData, List<VanguardDoc> docs)
+        {
+            if (historicData.Count == 0)
+            {
+                _log.WriteLog(LogLevel.Info, "No historicData to insert \n");
+                return;
+            }
+            for (var i = 0; i < historicData.Count; i++)
+            {
+                historicData[i].Data = ReplaceUnsupportedCharacters(historicData[i].Data, "'", ".");
+                InsertOcrData(connectString, historicData[i], docs[i]);
+            }
+        }
+
+        public string ReplaceUnsupportedCharacters(string word,string oldChar,string newChar)
+        {
+            return word.Contains(oldChar) ? word.Replace(oldChar, newChar) : word;
+        }
+
+        public void InsertOcrData(string connectString, HistoricalOcrData histoticData, VanguardDoc docs)
+        {
+            const string newErrorMessage = "0";
```

Comment:
what is the meaning of "0" error message? Isn't possible to pass an empty string or null instead(without creating of local constant)?